frontend/public/components/dashboard/dashboards-page/overview-dashboard/activity-card.tsx

```diff
@@ -24,20 +24,16 @@ import {
 import { uniqueResource } from './utils';
 import { PrometheusResponse } from '../../../graphs';
 import { connectToFlags, WithFlagsProps, FlagsObject } from '../../../../reducers/features';
+import { useK8sWatch } from '../../../utils/k8s-watch-hook';
 
-const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
+const eventsResource: FirehoseResource[] = [
+  { isList: true, kind: EventModel.kind, prop: 'events' },
+];
 
-const RecentEvent = withDashboardResources(
-  ({ watchK8sResource, stopWatchK8sResource, resources }) => {
-    React.useEffect(() => {
-      watchK8sResource(eventsResource);
-      return () => {
-        stopWatchK8sResource(eventsResource);
-      };
-    }, [watchK8sResource, stopWatchK8sResource]);
-    return <RecentEventsBody events={resources.events as FirehoseResult<EventKind[]>} />;
-  },
-);
+const RecentEvent = () => {
+  const resources = useK8sWatch(eventsResource);
+  return <RecentEventsBody events={resources.events as FirehoseResult<EventKind[]>} />;
```

Comment:
The need to typecast `resources.events` might indicate that `useK8sWatch` should return a more specific result type-wise.

Consider e.g. `useK8sWatch<FooKind>` or `useK8sWatch<FooKind | BarKind>` to assist with typing `resources.events` and consequently avoiding typecasts.

Reply:
ok, i removed the `R | R[]` part so the typecast is not needed anymore